diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/LineLength.kt

```diff
@@ -50,27 +50,23 @@ class LineLength : Rule("line-length") {
         node.text.lines().forEach {
             if (it.length > configuration.lineLength) {
                 val newNode = node.psi.findElementAt(offset + it.length - 1)!!.node
-                if (newNode.elementType == KDOC_TEXT)
-                    checkKDoc(newNode, configuration)
-                else
+                if (newNode.elementType != KDOC_TEXT || !isKDocValid(newNode)) {
                     LONG_LINE.warnAndFix(configRules, emitWarn, isFixMode,
                             "max line length ${configuration.lineLength}, but was ${it.length}",
                             offset.plus(node.startOffset)) {}
+                }
             }
             offset += it.length + 1
         }
     }
 
     // fixme json method
-    private fun checkKDoc(node: ASTNode, configuration: LineLengthConfiguration) {
-        if (node.text.length > configuration.lineLength) {
-            try {
-                URL(node.text).toURI()
-            } catch (e: IOException) {
-                LONG_LINE.warnAndFix(configRules, emitWarn, isFixMode,
-                        "max line length ${configuration.lineLength}, but was ${node.text.length}",
-                        node.startOffset) {}
-            }
+    private fun isKDocValid(node: ASTNode): Boolean {
+        return try {
+            URL(node.text).toURI()
```

Comment:
I think we should handle here not only the case with long URL occupying a whole line. For example, in KDocs people can wrap links in markdown. Checking for URL as a part of line using some simple regex could be more suitable.

@akuleshov7 please advice. I think it should be more clearly specified in codestyle guide.

Reply:
check the last word on URL